src/main/java/JavaQuestionOne/Main.java

```diff
@@ -0,0 +1,168 @@
+package JavaQuestionOne;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Scanner;
+
+public class Main {
+    Scanner sc = new Scanner(System.in);
+    public void fetchCLArgValues(String[] argument, ArrayList<Item> obj, Main mainObj ) throws Exception{
+        // Since user will be entering "-name Samsung Mobile -type raw -price 1000"
+        String[] args = argument;
+        do {
+            String name = null;
+            Integer quantity = null;
+            Double price = null;
+            String type = null;
+
+            HashMap<String, String> arguments = createMap(args);
+
+            // Fetch name
+            if (fetchNameFromInput(arguments)) {
+                name = arguments.get("-name");
+            }
+
+            // fetch Quantity
+            if (fetchQuantityFromInput(arguments)) {
+                quantity = Integer.parseInt(arguments.get("-quantity"));
+            } else {
+                quantity = 0;
+            }
+
+            // fetch price
+            if (fetchPriceFromInput(arguments)) {
+                price = Double.parseDouble(arguments.get("-price"));
+            } else {
+                price = 0.0;
+            }
+
+            // fetch type
+            if (fetchTypeFromInput(arguments)) {
+                type = arguments.get("-type");
+            } else {
+                throw new Exception("Item type is must");
+            }
+
+            // Create objects based on Type
+            if (type.equalsIgnoreCase("raw")) {
+                obj.add(new RawItem(name, quantity, price, type));
+            } else if (type.equalsIgnoreCase("manufactured")) {
+                obj.add(new ManufacturedItem(name, quantity, price, type));
+            } else if (type.equalsIgnoreCase("imported")) {
+                obj.add(new ImportedItem(name, quantity, price, type));
+            } else {
+                throw new Exception("Incorrect Type Passed");
```

Comment:
the flow will break on improper value of type. Shouldn't there be a retry mechanism?